src/arrayMethodJoin.js

```diff
@@ -5,8 +5,27 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(token = ',') {
+    let str = '';
+    let addToken = String(token);
+    if (token === null) {
+      addToken = String(null);
+    }
+    for (let i = 0; i < this.length; i++) {
+      if (
+        this[i] === null
+        || this[i] === undefined // eslint-disable-line
+      ) {
+        str += addToken;
+        continue;
```

Comment:
continue is hard to maintain. Rewrite for loop without it and without return. Just use conditions to add necessary parts to string, then return just once after the loop